### PR TITLE
Update Spack exercise

### PR DESCRIPTION
**Pull request by IshaanDesai**
## Description

This PR updates the Spack exercise for WiTe22-23.

## Checklist

- [x] I made sure that the markdown files are formatted properly.
- [x] I made sure that all hyperlinks are working.

### CODE REVIEW

File: 03_building_and_packaging/spack_exercise.md [hunk 1]
```diff
@@ -18,56 +18,29 @@ The [code repository](https://github.com/Simulation-Software-Engineering/spack-e
 - `v0.2.0` depends on Boost (at least version `1.65.1`)
 - `v0.3.0` depends on Boost (at least version `1.65.1`) and `yaml-cpp` (at least version `0.7.0`)
 
-Note that we require `yaml-cpp` in version `0.7.0` again. This package will be automatically installed via Spack if you specify the dependency correctly in you package. You should **not* install `yaml-cpp` manually in this exercise.
+**Note**: We require `yaml-cpp` in version `0.7.0` again. This package will be automatically installed via Spack if you specify the dependency correctly in you package. You should **not* install `yaml-cpp` manually in this exercise.
```
Comment (uekerman) [inline 03_building_and_packaging/spack_exercise.md:21]:
```suggestion
**Note**: We require `yaml-cpp` in version `0.7.0`. This package will be automatically installed via Spack if you specify the dependency correctly in you package. You should **not* install `yaml-cpp` manually in this exercise.
```

The "again" is a bit a leftover from last year.

File: 03_building_and_packaging/spack_exercise.md [hunk 2]
```diff
@@ -18,56 +18,29 @@ The [code repository](https://github.com/Simulation-Software-Engineering/spack-e
...
-
-You can find the recipe of the image inside the `docker/` directory of [the example code on GitHub](https://github.com/Simulation-Software-Engineering/spack-exercise). The image itself is based on Ubuntu 20.04 and has the Boost dependency preinstalled. Additionally, Spack has been set up such that you
-
- and two editors `vim` and `nano` preinstalled. If you want to install further software in your container you are free to do so.
+Please do all the development inside a Docker container. The Docker container is based on the image built from the recipe provided in the [exercise repository](https://github.com/Simulation-Software-Engineering/spack-exercise). You can find the recipe of the image inside the `docker/` directory of [the example code on GitHub](https://github.com/Simulation-Software-Engineering/spack-exercise). The image itself is based on Ubuntu 20.04 and has the Boost dependency preinstalled. Additionally, Spack has been set up in the recipe. Two editors `vim` and `nano` preinstalled. If you want to install further software in your container you are free to do so.
```
Comment (uekerman) [inline 03_building_and_packaging/spack_exercise.md:27]: Any specific reason why 20.04?

**Review by uekerman — APPROVED** [review]
Looks good. You have the file already in the timetable, right?

File: 03_building_and_packaging/spack_exercise.md (hunk 2 above)
Comment (IshaanDesai) [reply 03_building_and_packaging/spack_exercise.md:27]: With Ubuntu 22.04 there is a compiler conflict for `yaml-cpp` if used via Spack and without any additional compiler installations. I did not investigate this further in the interest of time, but will do so post-lecture.

**Pull request merged**